web/src/user_card_popover.js

```diff
@@ -579,6 +579,11 @@ function register_click_handlers() {
         toggle_user_card_popover_for_message(this, user, message.sender_id, true);
     });
 
+    // This does not cover drafts because drafts has a direct
+    // event listener which receives the event before our delegated
+    // event listener.
+    $("body").on("click", ".messagebox .user-mention", unsaved_message_user_mention_event_handler);
```

Comment:
It makes me uncomfortable that this click handler will run on the message feed too. 

I think maybe it'd be better to just have a single click handler set on body that looks at the parents of the `user-mention` to decide whether it's a normal message click (but not message-edit preview area) or not? That'd put all the logic for handling a mention click in one place.

Reply:
I tried this out where I set the click handler for a normal message user mention on body too. In that case, clicking on a user-mention opens up the compose box since that handler is called first. [That handler](https://github.com/zulip/zulip/blob/69194489177bc3b43ff7002bd4de92ceda628f77/web/src/click_handlers.js#L217) is defined on `#main_div` and will capture the event before a click handler attached to body. 

We can probably do some messing around with the current click handler for opening the compose box in order to unify the logic for `user_mention` but not sure if it will be a net improvement or not.

Reply:
OK, this can probably be fine but it needs a good comment.